src/server/coprocessor/executor/selection.rs

```diff
@@ -0,0 +1,269 @@
+// limitations under the License.
+
+// FIXME(andelf): remove following later
+#![allow(dead_code)]
+
+use std::rc::Rc;
+use std::collections::HashSet;
+
+use tipb::executor::Selection;
+use tipb::schema::ColumnInfo;
+use tipb::expression::{Expr, ExprType};
+use util::xeval::{Evaluator, EvalContext};
+use util::codec::number::NumberDecoder;
+
+use super::{Row, Executor};
+use super::super::Result;
+use super::super::endpoint::inflate_with_col;
+
+/// Collect all mentioned column ids in expr tree
+pub struct ExprColumnRefVisitor {
+    pub column_ids: HashSet<i64>,
+}
+
+impl ExprColumnRefVisitor {
+    pub fn new() -> ExprColumnRefVisitor {
+        ExprColumnRefVisitor { column_ids: HashSet::new() }
+    }
+
+    pub fn visit_expr<'e>(&mut self, expr: &'e Expr) -> Result<()> {
+        if expr.get_tp() == ExprType::ColumnRef {
+            self.column_ids.insert(try!(expr.get_val().decode_i64()));
+        } else {
+            for sub_expr in expr.get_children() {
+                try!(self.visit_expr(sub_expr));
+            }
+        }
+        Ok(())
+    }
+}
+
+pub struct SelectionExecutor<'a> {
+    conditions: Vec<Expr>,
+    columns: Vec<ColumnInfo>,
+    ctx: Rc<EvalContext>,
+    src: Box<Executor + 'a>,
+}
+
+impl<'a> SelectionExecutor<'a> {
+    pub fn new(mut meta: Selection,
+               ctx: Rc<EvalContext>,
+               columns_info: &[ColumnInfo],
+               src: Box<Executor + 'a>)
+               -> Result<SelectionExecutor<'a>> {
+        let conditions = meta.take_conditions().into_vec();
+
+        let mut visitor = ExprColumnRefVisitor::new();
+        for cond in &conditions {
+            try!(visitor.visit_expr(cond));
+        }
+
+        // FIXME(andelf): assume all items in columns_info are unique.
+        //   For now, `SelectionExecutor` is dangling.
+        let columns = columns_info.iter()
+            .filter(|col| visitor.column_ids.get(&col.get_column_id()).is_some())
+            .cloned()
+            .collect::<Vec<ColumnInfo>>();
+
+        assert_eq!(columns.len(), visitor.column_ids.len());
+
+        Ok(SelectionExecutor {
+            conditions: conditions,
+            columns: columns,
+            ctx: ctx,
+            src: src,
+        })
+    }
+}
+
+
+impl<'a> Executor for SelectionExecutor<'a> {
+    fn next(&mut self) -> Result<Option<Row>> {
+        'next: loop {
+            if let Some(row) = try!(self.src.next()) {
+                let mut evaluator = Evaluator::default();
+                try!(inflate_with_col(&mut evaluator,
+                                      &self.ctx,
+                                      &row.data,
+                                      &self.columns,
+                                      row.handle));
+                for expr in &self.conditions {
+                    let is_selected =
+                        evaluator.eval(&self.ctx, expr)?.into_bool(&self.ctx)?.unwrap_or(false);
```

Comment:
Why not use `try!`(I guess the style of 'tikv' is to use `try!` instead of `?`)? 
And could we split  
```evaluator.eval(&self.ctx, expr)?.into_bool(&self.ctx)?.unwrap_or(false);```
into two independent lines like `endpoint.rs` did? 
I  guess that may makes it  more easy to read?